Fix TestHttp.test_html

Currently this test is disabled (with `@pytest.mark.xfail`) because it fails non-deterministically. If I remove the pytest mark and run it, I sometimes get:

```
$ tox -- tests/checker/test_http.py
...
=================================== FAILURES ===================================
______________________________ TestHttp.test_html ______________________________

self = <tests.checker.test_http.TestHttp testMethod=test_html>

    def test_html (self):
        confargs = dict(recursionlevel=1)
>       self.file_test("http.html", confargs=confargs)

tests/checker/test_http.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/checker/__init__.py:199: in file_test
    self.fail_unicode(msg)
tests/checker/__init__.py:205: in fail_unicode
    self.fail(msg)
E   AssertionError: http://localhost:39497/tests/checker/data/http.html
E   --- 
E   
E   +++ 
E   
E   @@ -39,13 +39,13 @@
E   
E    real url http://localhost:39497/?q=%C3%BC
E    name html entities
E    valid
E   +url file.css
E   +cache key http://localhost:39497/tests/checker/data/file.css
E   +real url http://localhost:39497/tests/checker/data/file.css
E   +valid
E    url file.html
E    cache key http://localhost:39497/tests/checker/data/file.html
E    real url http://localhost:39497/tests/checker/data/file.html
E   -valid
E   -url file.css
E   -cache key http://localhost:39497/tests/checker/data/file.css
E   -real url http://localhost:39497/tests/checker/data/file.css
E    valid
E    url file.asc
E    cache key http://localhost:39497/tests/checker/data/file.asc
====================== 1 failed, 1 passed in 3.96 seconds ======================
ERROR: InvocationError: '/home/mg/src/linkchecker/.tox/py27/bin/py.test tests/checker/test_http.py'
___________________________________ summary ____________________________________
ERROR:   py27: commands failed
```

AFAICT this is because LinkChecker crawls HTML attributes in a non-deterministic order, so when it encounters

```
<!-- multiple links in one tag -->
<applet archive="file.html" src="file.css">
```

there's no saying if it'll visit `file.html` or `file.css` fix.

Proposed approach: find where the crawling is done and sort HTML attributes alphabetically before crawling them.

--- a/linkcheck/htmlutil/linkparse.py
+++ b/linkcheck/htmlutil/linkparse.py
@@ -208,7 +208,7 @@ class LinkFinder (TagFinder):
             self.base_ref = attrs.get_true("href", u'')
         tagattrs = self.tags.get(tag, self.universal_attrs)
         # parse URLs in tag (possibly multiple URLs in CSS styles)
-        for attr in tagattrs.intersection(attrs):
+        for attr in sorted(tagattrs.intersection(attrs)):
             if tag == "meta" and not is_meta_url(attr, attrs):
                 continue
             if tag == "form" and not is_form_get(attr, attrs):

--- a/tests/checker/test_http.py
+++ b/tests/checker/test_http.py
@@ -29,7 +29,6 @@ class TestHttp (HttpServerTest):
         super(TestHttp, self).__init__(methodName=methodName)
         self.handler = CookieRedirectHttpRequestHandler
 
-    @pytest.mark.xfail(reason="fails non-deterministically")
     def test_html (self):
         confargs = dict(recursionlevel=1)
         self.file_test("http.html", confargs=confargs)

